proot-distro.sh

```diff
@@ -1643,15 +1643,10 @@ command_login() {
 		set -u
 	done
 	unset var
-
-	local -a login_shell_args
+ 
 	if [ $# -ge 1 ]; then
-		# Wrap in quotes each argument to prevent word splitting.
-		local -a login_shell_args
-		for i in "$@"; do
-			login_shell_args+=("'$i'")
-		done
-		set -- "-c" "${login_shell_args[*]}"
+		# Escape each argument to prevent word splitting.
+		set -- "-c" "$(printf " %q" "$@")"
```

Comment:
Big +1 for using `%q`[^1]
```zig
// man 1 bash
%q     causes printf to output the corresponding argument in a
       format that can be reused as shell input.
```

Another alternative that avoids the need for a command substitution
would have been to use `${@@Q}`[^2].
Which is admittedly sort of weird syntax.
But quoting from the man page again:
```zig
${parameter@operator}
Parameter transformation.  The expansion is either a
transformation of the value of parameter or information about
parameter itself, depending on the value of operator.  Each
operator is a single letter:

// [...]

Q      The expansion is a string that is the value of parameter
       quoted in a format that can be reused as input.

// [...]

If parameter is @ or *, the operation is applied to each
positional parameter in turn, and the expansion is the resultant
list.  If parameter is an array variable subscripted with @ or
*, the operation is applied to each member of the array in turn,
and the expansion is the resultant list.

The result of the expansion is subject to word splitting and
pathname expansion as described below.
```
I don't think it's a big enough deal to go and fix it now after the fact,
`printf %q` does the same job after all, but the `${parameter@operator}` expansions are good to have in the back of the head while writing Bash.

[^1]: https://www.man7.org/linux/man-pages/man1/bash.1.html#SHELL_BUILTIN_COMMANDS:~:text=status%20is%200.-,printf,-%5B%2Dv%20var
[^2]: https://www.man7.org/linux/man-pages/man1/bash.1.html#EXPANSION:~:text=%24%7Bparameter%40operator%7D